### PR TITLE
=str Reduce the internal buffer usage of MergePrioritized.

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/25823

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala [hunk 1]
```diff
@@ -329,90 +328,94 @@ object MergePrioritized {
...
 
-      private def dequeueAndDispatch(): Unit = {
-        val in = selectNextElement()
-        push(out, grab(in))
-        if (upstreamsClosed && !hasPending) completeStage() else tryPull(in)
-      }
+      private def upstreamsClosed: Boolean = runningUpstreams == 0
 
-      private def selectNextElement() = {
+      private def select(): Inlet[T] = {
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala:390]: This mehtod is not select a next element but a `Inlet` .

File: akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala [hunk 2]
```diff
@@ -329,90 +328,94 @@ object MergePrioritized {
...
 
-  val in: immutable.IndexedSeq[Inlet[T]] = Vector.tabulate(inputPorts)(i => Inlet[T]("MergePrioritized.in" + i))
+  val in: immutable.IndexedSeq[Inlet[T]] = Vector.tabulate(priorities.size)(i => Inlet[T]("MergePrioritized.in" + i))
   val out: Outlet[T] = Outlet[T]("MergePrioritized.out")
   override def initialAttributes: Attributes = DefaultAttributes.mergePrioritized
   override val shape: UniformFanInShape[T, T] = UniformFanInShape(out, in: _*)
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new GraphStageLogic(shape) with OutHandler {
-      private val allBuffers = Vector.tabulate(priorities.size)(i => FixedSizeBuffer[Inlet[T]](priorities(i)))
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala:343]: I think this buffer is not needed and it's not fully used too.

**Comment by hepin1989** [comment]
@andersstorhaug @arpanchaudhury Frinendly ping

**Review by jrudolph — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala [hunk 3]
```diff
@@ -329,90 +328,94 @@ object MergePrioritized {
...
-        var r = randomGen.nextInt(tp)
-        var next: Inlet[T] = null
-        ix = 0
-
-        while (ix < in.size && next == null) {
-          if (allBuffers(ix).nonEmpty) {
-            r -= priorities(ix)
-            if (r < 0) next = allBuffers(ix).dequeue()
+        if (tp == 0) {
+          //all inlets are not available
```
[the remark this answers is not in the document]
Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala]: thanks

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**